src/http2/index.ts

```
@@ -0,0 +1,338 @@
+import http2                  from 'http2';
+import { EventEmitter }       from 'events';
+import { v4 as uuid }         from 'uuid';
+import { Application_, Json } from '../core2/application';
+
+type StreamListener = (stream: http2.Http2Stream, headers: http2.IncomingHttpHeaders) => void;
+
+/**
+ * creates and manages our collection of servers
+ */
+
+class Http2FactorySingleton {
+  private _app: Application_ | undefined;
+  private _servers: { [key: string]: Http2Server } = {};
+
+  get app() {
+    return this._app;
+  }
+
+  get emitter() {
+    return this.app?.emitter;
+  }
+
+  public Create(name: string) {
+    this.ValidateApp();
+    const newServer = new Http2Server(name, false);
+    this.AddServer(name, newServer);
+    return newServer;
+  }
+
+  public CreateSecure(name: string, key: Buffer, cert: Buffer) {
+    this.ValidateApp();
+    const newServer = new Http2Server(name, true, key, cert);
+    this.AddServer(name, newServer);
+    return newServer;
+  }
+
+  public Use(app: Application_) {
+    this._app = app;
+  }
+
+  public GetServer(serverName: string): Http2Server {
```

serverName is literally the ugliest param I ever read. No like. Good method, terrible fucking param name bro. <3

---

I noticed you are using get server in decorators, heads up, decorators swallow errors. I have been thinking of ways to solve this for a while, and really the easiest most direct way I can think of is make an error stack. The filly decorators type called should do the check, print the error, and process exit

---

or if you wanna try catch in the decorators

---

gotcha, I will work on this